server/src/main/java/org/apache/uniffle/server/ShuffleTaskInfo.java

```diff
@@ -198,6 +201,25 @@ public Set<Integer> getShuffleIds() {
     return partitionDataSizes.keySet();
   }
 
+  public void incrBlockNumber(int shuffleId, int partitionId) {
+    this.partitionBlockCounters
+        .computeIfAbsent(shuffleId, x -> JavaUtils.newConcurrentMap())
+        .computeIfAbsent(partitionId, x -> new AtomicLong())
+        .incrementAndGet();
+  }
+
+  public long getBlockNumber(int shuffleId, int partitionId) {
+    Map<Integer, AtomicLong> partitionBlockCounters = this.partitionBlockCounters.get(shuffleId);
+    if (partitionBlockCounters == null) {
```

Comment:
I think we can use `getOrDefault` to simplify the following codes.

Reply:
I don't introduce extra the object creating cost.

Reply:
OK